pydeseq2/dds.py

```diff
@@ -522,16 +510,11 @@ def fit_LFC(self):
         """
 
         # Check that MAP dispersions are available. If not, compute them.
-        if not hasattr(self, "dispersions"):
+        if "dispersions" not in self.varm:
             self.fit_MAP_dispersions()
 
-        # Exclude genes with all zeroes
-        counts_nonzero = self.counts.loc[:, self.non_zero_genes].values
-        num_genes = counts_nonzero.shape[1]
-
-        X = self.design_matrix.values
-        size_factors = self.size_factors.values
-        disps = self.dispersions.loc[self.non_zero_genes].values
+        # Convert design matrix to numpy for speed
+        X = self.obsm["design_matrix"].values
```

Comment:
same comment, consider renaming `X` as `design_matrix_np`?

Reply:
done in [2bcc4dc](https://github.com/owkin/PyDESeq2/pull/66/commits/2bcc4dc7abc6d41951f4dcc70263b78a2b44ffdc)